plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatActionPromoter.kt

```diff
@@ -0,0 +1,32 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cwc.inline
+
+import com.intellij.openapi.actionSystem.ActionPromoter
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.keymap.KeymapUtil
+import com.intellij.openapi.util.SystemInfo
+
+class InlineChatActionPromoter : ActionPromoter {
+    override fun promote(actions: MutableList<out AnAction>, context: DataContext): MutableList<AnAction> {
```

Comment:
if the context is not appropriate then this should not trigger either

Reply:
could you pls elaborate under which context it shouldn't trigger?

Reply:
I can't think of a case as long as they have amazon q extension installed

Reply:
for example, if the action is not triggered from an editor

Reply:
ok will add that

Reply:
done